isolate.sh

```diff
@@ -825,6 +832,121 @@ function wait_for_nss_exist() {
     wait_for_condition "${condition}" ${retries} ${sleep_time} "${wait_message}" "${success_message}" "${error_message}"
 }
 
+function patch_cs_cr_for_management_ingress() {
+    title "Updating commonservice common-service in namespace ${MASTER_NS} for management ingress CR ..."
+    "${OC}" get commonservice common-service -n "${MASTER_NS}" -o yaml > tmp_cs_cr.yaml
+
+    local is_exist_in_cs=$("${YQ}" eval '.spec.services[].name' tmp_cs_cr.yaml | grep "ibm-management-ingress-operator" || echo "false")
+    if [[ "${is_exist_in_cs}" == "false" ]]; then
+        "${YQ}" -i eval '.spec.services += [{"name": "ibm-management-ingress-operator", "spec": {"managementIngress": {"multipleInstancesEnabled": false}}}]' tmp_cs_cr.yaml
+    else
+        info "ibm-management-ingress-operator already exists in CS CR .spec.services, updating it ..."
+        "${YQ}" -i eval '(.spec.services[] |= select(.name == "ibm-management-ingress-operator").spec.managementIngress.multipleInstancesEnabled = false)' tmp_cs_cr.yaml
+    fi
+  
+    local retries=10
+    local sleep_time=5
+    local total_time_mins=$(( sleep_time * retries / 60))
+    local wait_message="Waiting for commonservice common-service in namespace ${MASTER_NS} to be updated for management ingress CR ..."
+    local success_message="Commonservice common-service updated in ${MASTER_NS} for management ingress CR."
+    local error_message="Timeout after ${total_time_mins} minutes waiting for commonservice common-service to be updated for management ingress CR."
+    while true; do
+        if [[ ${retries} -eq 0 ]]; then
+            rm -f tmp_cs_cr.yaml
+            error "${error_message}"
+        fi
+
+        local result=$("${OC}" apply -n "${MASTER_NS}" -f tmp_cs_cr.yaml 2>&1 || echo "fail")
+        if [[ "${result}" == "fail" ]]; then
+            retries=$(( retries - 1 ))
+            info "RETRYING: ${wait_message} (${retries} left)"
+            sleep ${sleep_time}
+        else
+            success "${success_message}"
+            break
+        fi
+    done
+    rm -f tmp_cs_cr.yaml
+}
+
+function patch_opconfig_for_management_ingress() {
+    title "Updating operandconfig common-service in namespace ${MASTER_NS} for management ingress CR ..."
+    "${OC}" get operandconfig common-service -n "${MASTER_NS}" -o yaml > tmp_oc_cr.yaml
+
+    local is_exist_in_opcon=$("${YQ}" eval '.spec.services[].name' tmp_oc_cr.yaml | grep "ibm-management-ingress-operator" || echo "false")
+    if [[ "${is_exist_in_opcon}" == "false" ]]; then
+        "${YQ}" -i eval '.spec.services += [{"name": "ibm-management-ingress-operator", "spec": {"managementIngress": {"multipleInstancesEnabled": false}}}]' tmp_oc_cr.yaml
+    else
+        info "ibm-management-ingress-operator already exists in operandconfig CR .spec.services, updating it ..."
```

Comment:
Under which condition mgmt ingress entry is already in OperandConfig here, vs is not present there?

Reply:
Typically mgmt ingress entry will always be present in OperandConfig, so we won't need to worry about this.
It is only used to handle the corner case that user accidentally changes the OperandConfig, and cause some information missing.